src/steps/check-lighthouse-performance.ts

```diff
@@ -0,0 +1,56 @@
+import { BaseStep, Field, StepInterface } from '../core/base-step';
+import { Step, RunStepResponse, FieldDefinition, StepDefinition } from '../proto/cog_pb';
+
+export class CheckLighthousePerformance extends BaseStep implements StepInterface {
+
+  protected stepName: string = 'Check a page\'s Lighthouse performance score';
+  // tslint:disable-next-line:max-line-length
+  protected stepExpression: string = 'the (?<throttleTo>mobile|desktop) lighthouse performance score should be (?<expectedScore>\\d{1,3}) or higher';
+  protected stepType: StepDefinition.Type = StepDefinition.Type.VALIDATION;
+  protected expectedFields: Field[] = [{
+    field: 'throttleTo',
+    type: FieldDefinition.Type.STRING,
+    description: 'Throttle Level (mobile or desktop)',
+  }, {
+    field: 'expectedScore',
+    type: FieldDefinition.Type.NUMERIC,
+    description: 'Expected Score',
+  }];
+
+  async executeStep(step: Step): Promise<RunStepResponse> {
+    const stepData: any = step.getData().toJavaScript();
+    const throttleTo: 'mobile' | 'desktop' = stepData.throttleTo;
+    const expectedScore: number = stepData.expectedScore;
+
+    try {
+      const currentUrl = await this.client.getCurrentPageInfo('url');
+      const lhr = await this.client.getLighthouseScores(currentUrl.toString(), throttleTo);
+
+      const performance: any = Object.values(lhr.categories)[0];
+      const actualScore = performance.score * 100;
+      if (actualScore < expectedScore) {
+        const audits = Object.values(lhr.audits);
+        console.log(audits.filter((audit: any) => audit.details && audit.details.type === 'opportunity'));
+        return this.fail('The page\'s performance score of %d was lower than the expected score of %d.\n\n\n Opportunities for improvement:\n%s', [
```

Comment:
- [ ] After playing with this some locally, I found myself missing the throttle-level in the messages. Perhaps we add it in like: `The page\'s %s performance score of...` where `%s` is `throttleTo`